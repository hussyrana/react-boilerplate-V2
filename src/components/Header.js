import React from 'react';
import { connect } from 'react-redux';
import { Link} from 'react-router-dom';
import { startLogout } from '../actions/auth';


export const Header = ({startLogout}) => (
    <header className="header">
        <div className="content-container">
            <div className="header-content">
                <Link className="header-title" to="/dashboard" >
                     <h1>Boilerpalte-v2</h1>
                </Link>
                <button className="login-button button--link" onClick={startLogout}>Logout</button>
            </div>
        </div>
    </header>
);

const mapDispatchToProps = (dispatch)=>({
    startLogout: () => dispatch(startLogout())
});


export default connect(undefined, mapDispatchToProps)(Header);