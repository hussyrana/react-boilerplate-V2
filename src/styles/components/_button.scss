.login-button{
    background:$blue;
    color:white;
    padding: $s-size;
    display: inline-block;
    font-weight: 300;
    font-size: $font-size-lg;
    line-height: 1;
    border:none;
    text-decoration: none;
}
.button--link{
    background: none;
}
.login-button--remove{
    background: #888;
    margin-bottom: $l-size;
}