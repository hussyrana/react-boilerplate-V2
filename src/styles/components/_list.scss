.list-header{
    background: $page-header-bg;
    border: 1px solid darken($page-header-bg, 7%);
    color: $grey;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: $s-size $m-size;
}
.list-item{
    display: flex;
    flex-direction: column;
    text-decoration: none;
    border-top: none;
    color: $dark-grey;
    padding: $s-size;
    border: 1px solid darken($page-header-bg, 7%);
    transition: background .3s ease;
    &:hover {
        background: $page-header-bg;
    }
    @media (min-width: $desktop-breakpoint){ 
      flex-direction:row;
      align-items: center;
        justify-content: space-between;
        padding: 0 $m-size;
    }
}
.list-body{
    margin-bottom: $m-size;
    @media (min-width: $desktop-breakpoint){ 
        margin-bottom: $l-size;
      }
}

.list-item__message{
    align-items: center;
    justify-content: center;
    padding: $s-size;
    color:$grey;
    &:hover {
        background: none;
    }
}

.list-item__title{
margin: 0;
word-break: break-all;
}
.list-item__Subtitle{
    color:$grey;
    font-size: $font-size-sm;
}
.list-item__data{
    margin: $s-size 0 0 0;
    @media (min-width: $desktop-breakpoint){ 
          margin: 0;
          padding-left: $s-size;
      }
}