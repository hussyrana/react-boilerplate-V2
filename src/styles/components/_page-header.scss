.page-header{
   background:$page-header-bg;
   padding: $l-size 0;
   margin-bottom: $s-size; 
}
.page-header__title{
    margin:0;
    font-weight: 300;
    span {
        font-weight: 700;
    }
}
.page-header__actions{
    margin-top: $m-size;
}