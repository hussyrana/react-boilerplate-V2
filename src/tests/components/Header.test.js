import React from 'react';
import {Header} from '../../components/Header';
import {shallow} from 'enzyme';

test('render Header correctly', () => {
    const warpper = shallow(<Header startLogout = {()=>{}}/>);
    expect(warpper).toMatchSnapshot();
});

test('should call startLogout on Button click', ()=>{
    const startLogout = jest.fn();
    const wrapper = shallow(<Header startLogout={startLogout}/>)
    wrapper.find('button').simulate('click');
    expect(startLogout).toHaveBeenCalled();
});
