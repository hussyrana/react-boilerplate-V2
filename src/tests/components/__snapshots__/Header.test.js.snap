// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Header correctly 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header-content"
    >
      <Link
        className="header-title"
        to="/dashboard"
      >
        <h1>
          Boilerpalte-v2
        </h1>
      </Link>
      <button
        className="login-button button--link"
        onClick={[Function]}
      >
        Logout
      </button>
    </div>
  </div>
</header>
`;
